common/src/main/java/com/microsoft/identity/common/internal/util/StringUtil.java

```diff
@@ -153,4 +154,25 @@ public static int compareSemanticVersion(final String thisVersion, final String
 
         return 0;
     }
+
+    /**
+     * Counts the number of occurrences of one String in another.
+     *
+     * @param str
+     * @param subString
+     * @return int
+     */
+    public static int countMatches(@NonNull String str, @Nullable String subString) {
+        int count = 0;
+        if (StringUtil.isEmpty(str) || StringUtil.isEmpty(subString)) {
+            return count;
+        }
+        for(int i = 0; i < (str.length() - subString.length()); i++) {
```

Comment:
There is an off by one error here. Need `less than or equal to comparison <=` to get accurate count